CMakeLists.txt

```diff
@@ -417,37 +393,35 @@ if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU" OR
 
     # C++
     set(CMAKE_CXX_FLAGS_DEBUG ${BUILD_CXX_FLAGS_DEBUG}
-        CACHE STRING "Can't change here -- see BUILD_CXX..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
     set(CMAKE_CXX_FLAGS_RELEASE ${BUILD_CXX_FLAGS_RELEASE}
-        CACHE STRING "Can't change here -- see BUILD_CXX..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
     set(CMAKE_CXX_FLAGS_RELWITHDEBINFO ${BUILD_CXX_FLAGS_RELWITHDEBINFO}
-        CACHE STRING "Can't change here -- see BUILD_CXX..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
     set(CMAKE_CXX_FLAGS_MINSIZEREL ${BUILD_CXX_FLAGS_MINSIZEREL}
-        CACHE STRING "Can't change here -- see BUILD_CXX..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
```

Comment:
I don't think the machinery here is proper/correct for recent CMake's, and/or should probably be better scoped (i.e. set as target properties).

Reply:
I reverted these INTERNAL settings as these variables should definitely not be set to INTERNAL. But, the flag handling here is generally not ideal (and inconsistent across different platforms) since it overrides any manually specified flags on the command line. (Also the BUILD_CXX variable is not set or used anywhere so the docstrings are incorrect.)